src/index.js

```diff
@@ -200,6 +204,72 @@ class IpfsRepo {
   exists (callback) {
     this.version.exists(callback)
   }
+
+  /**
+   * Get repo status.
+   *
+   * @param {function(Error, Object)} callback
+   * @return {void}
+   */
+  stat (callback) {
+    waterfall([
+      (cb) => cb(null, {
+        repoPath: this.path,
+        repoSize: new Big(0),
+        storageMax: new Big(noLimit)
+      }),
+      (stats, cb) => this.version.get((err, version) => {
+        stats.version = version
+        cb(err, stats)
+      }),
+      (stats, cb) => this.blocks.query({}, (err, list) => {
+        list = list || []
+        stats.numObjects = new Big(list.length)
+
+        list.forEach(block => {
+          stats.repoSize = stats.repoSize
+            .plus(block.value.byteLength)
+            .plus(block.key._buf.byteLength)
+        })
+
+        cb(err, stats)
+      }),
+      (stats, cb) => getSize(this.datastore, (err, size) => {
+        stats.repoSize = stats.repoSize.plus(size)
+        cb(err, stats)
+      }),
+      (stats, cb) => getSize(this.keys, (err, size) => {
+        stats.repoSize = stats.repoSize.plus(size)
+        cb(err, stats)
+      }),
+      (stats, cb) => this.config.get('Datastore.StorageMax', (err, max) => {
+        if (!err) {
+          stats.storageMax = new Big(max)
+        }
+
+        cb(null, stats)
+      })
+    ], callback)
+  }
+}
+
+function getSize (queryFn, callback) {
+  pull(
+    queryFn.query({}),
```

Comment:
Try to avoid buffering: I think you can reduce this iteratively instead of buffering the entire list.

Reply:
My comment refers to line below where you do `pull.collect`.

Reply:
Sorry, but I didn't get this. How can I do this iteratively?

Reply:
Since the query is a pull-stream, you can use `pull.reduce` like this:

```js
pull(
  source,
  pull.reduce((val) => compute(val), 0, callback))
```

Reply:
Oh, didn't know that. Thanks

Reply:
Done! 😄